packages/p2p/src/components/advertiser-page/advertiser-page.jsx

```diff
@@ -0,0 +1,278 @@
+import { Button, Dialog, Icon, Loading, Popover, Table, Tabs } from '@deriv/components';
+import { generateHexColourFromNickname, getShortNickname } from 'Utils/string';
+import { height_constants } from 'Utils/height_constants';
+import Dp2pContext from 'Components/context/dp2p-context';
+import { InfiniteLoaderList } from 'Components/table/infinite-loader-list.jsx';
+import { requestWS } from 'Utils/websocket';
+import Empty from 'Components/empty/empty.jsx';
+import Popup from '../buy-sell/popup.jsx';
+import { buy_sell } from '../../constants/buy-sell';
+import { localize } from '../i18next';
+import './advertiser-page.scss';
+
+const RowComponent = React.memo(({ data, is_buy_advert, showAdPopup, style }) => {
+    const { advertiser_id } = React.useContext(Dp2pContext);
+    const is_my_ad = data.advertiser_id === advertiser_id;
+
+    return (
+        <div style={style}>
+            <Table.Row className='advertiser-page__adverts-table_row'>
+                <Table.Cell>
+                    {data.display_min_available}&ndash;{data.display_max_available} {data.offer_currency}
+                </Table.Cell>
+                <Table.Cell className='advertiser-page__adverts-price'>
+                    {data.display_price_rate} {data.transaction_currency}
+                </Table.Cell>
+                {is_my_ad ? (
+                    <Table.Cell />
+                ) : (
+                    <Table.Cell className='advertiser-page__adverts-button'>
+                        <Button primary small onClick={() => showAdPopup(data)}>
+                            {is_buy_advert ? localize('Buy') : localize('Sell')} {data.offer_currency}
+                        </Button>
+                    </Table.Cell>
+                )}
+            </Table.Row>
+        </div>
+    );
+});
+
+RowComponent.displayName = 'RowComponent';
+
+const AdvertiserPage = ({ navigate, selected_ad, showVerification }) => {
+    const { is_advertiser } = React.useContext(Dp2pContext);
+    const { advertiser_name, advertiser_id } = selected_ad;
+    const [active_index, setActiveIndex] = React.useState(0);
+    const [ad, setAd] = React.useState(null);
+    const [adverts, setAdverts] = React.useState([]);
+    const [counterparty_type, setCounterpartyType] = React.useState(buy_sell.BUY);
+    const [error_message, setErrorMessage] = React.useState('');
+    const [has_adverts, setHasAdverts] = React.useState(false);
+    const height_values = [
+        height_constants.screen,
+        height_constants.core_header,
+        height_constants.page_overlay_header,
+        height_constants.page_overlay_content_padding,
+        height_constants.tabs,
+        height_constants.filters,
+        height_constants.filters_margin,
+        height_constants.table_header,
+        height_constants.core_footer,
+    ];
+    const item_height = 56;
+    const [is_loading, setIsLoading] = React.useState(true);
+    const [is_mounted, setIsMounted] = React.useState(false);
```

Comment:
- I suggest using `React.useRef` here to keep track of whether the component is mounted so we don't unnecessarily re-render when unmounting this component.
- Since this is a common pattern, you could create a reusable `useIsMounted` hook for this in `@deriv/shared`.